types/abci.go

```diff
@@ -23,6 +23,10 @@ type ProcessProposalHandler func(Context, *abci.ProcessProposalRequest) (*abci.P
 // PrepareProposalHandler defines a function type alias for preparing a proposal
 type PrepareProposalHandler func(Context, *abci.PrepareProposalRequest) (*abci.PrepareProposalResponse, error)
 
+// CheckTxHandler defines a function type alias for executing logic before transactions are executed.
+// `RunTx` is a function type alias for executing logic before transactions are executed.
```

Comment:
I would document that this does not actually override ant ehandler checks and such are performed.

Reply:
could you also explain a little why do we pass a wrapper of runTx here? do we want to run checkTx handler even if the TX fails run Tx?

Reply:
we want people to avoid changing the execuiton mode of checktx, this way only the transaction is passed in

Reply:
added this to the godocs